pin/pin.go

```diff
@@ -259,90 +269,99 @@ func (p *pinner) isPinnedWithType(k key.Key, mode PinMode) (string, bool, error)
 			return "", false, err
 		}
 		if has {
-			return rk.B58String(), true, nil
+			return rc.String(), true, nil
 		}
 	}
 	return "", false, nil
 }
 
-func (p *pinner) CheckIfPinned(keys ...key.Key) ([]Pinned, error) {
+func (p *pinner) CheckIfPinned(cids ...*cid.Cid) ([]Pinned, error) {
 	p.lock.RLock()
 	defer p.lock.RUnlock()
-	pinned := make([]Pinned, 0, len(keys))
-	toCheck := make(map[key.Key]struct{})
+	pinned := make([]Pinned, 0, len(cids))
+	toCheck := cid.NewSet()
 
 	// First check for non-Indirect pins directly
-	for _, k := range keys {
-		if p.recursePin.HasKey(k) {
-			pinned = append(pinned, Pinned{Key: k, Mode: Recursive})
-		} else if p.directPin.HasKey(k) {
-			pinned = append(pinned, Pinned{Key: k, Mode: Direct})
-		} else if p.isInternalPin(k) {
-			pinned = append(pinned, Pinned{Key: k, Mode: Internal})
+	for _, c := range cids {
+		if p.recursePin.Has(c) {
+			pinned = append(pinned, Pinned{Key: c, Mode: Recursive})
+		} else if p.directPin.Has(c) {
+			pinned = append(pinned, Pinned{Key: c, Mode: Direct})
+		} else if p.isInternalPin(c) {
+			pinned = append(pinned, Pinned{Key: c, Mode: Internal})
 		} else {
-			toCheck[k] = struct{}{}
+			toCheck.Add(c)
 		}
 	}
 
 	// Now walk all recursive pins to check for indirect pins
-	var checkChildren func(key.Key, key.Key) error
-	checkChildren = func(rk key.Key, parentKey key.Key) error {
+	var checkChildren func(*cid.Cid, *cid.Cid) error
+	checkChildren = func(rk, parentKey *cid.Cid) error {
 		parent, err := p.dserv.Get(context.Background(), parentKey)
 		if err != nil {
 			return err
 		}
 		for _, lnk := range parent.Links {
-			k := key.Key(lnk.Hash)
+			c := cid.NewCidV0(lnk.Hash)
```

Comment:
I assume this is explicitly v0 because we don't wanna start using v1 right now already